### PR TITLE
add session credentials as authorization method

### PR DESCRIPTION
**Pull request by rachelspark**
Design doc: https://www.notion.so/modal-com/Code-Execution-Playground-for-Modal-Examples-14b781084a184c8182e10627e27558bf?pvs=4#a6ce613bd667437796df660c6bb2a79f

### CODE REVIEW

File: modal/client.py [hunk 1]
```diff
@@ -155,18 +173,18 @@ async def _init(self):
...
+            #     req,
+            #     attempt_timeout=CLIENT_CREATE_ATTEMPT_TIMEOUT,
+            #     total_timeout=CLIENT_CREATE_TOTAL_TIMEOUT,
+            # )
+            # if resp.warning:
+            #     ALARM_EMOJI = chr(0x1F6A8)
+            #     warnings.warn(f"{ALARM_EMOJI} {resp.warning} {ALARM_EMOJI}", DeprecationError)
             self._authenticated = True
-            self.image_builder_version = resp.image_builder_version
+            self.image_builder_version = "2024.04"
```
Comment (mwaskom) [inline modal/client.py:187]: FYI you can force this with `MODAL_IMAGE_BUILDER_VERSION` when you run things

**Comment by mwaskom** [comment]
What's the difference between the "session credentials" usecase and the usecase supported by `Client.from_credentials`?

**Comment by rachelspark** [comment]
> What's the difference between the "session credentials" usecase and the usecase supported by `Client.from_credentials`?

This is a bit of a hacky way that we're authorizing sandboxes to be run for the code playground without having to create another API token. See discussion in the design doc: https://www.notion.so/modal-com/Code-Execution-Playground-for-Modal-Examples-14b781084a184c8182e10627e27558bf?pvs=4#a6ce613bd667437796df660c6bb2a79f.

**Comment by mwaskom** [comment]
Closing as stale — I gather we didn't need this for the Playground.

**Pull request closed without merging**